slides/src/content/unit-testing.md

```diff
@@ -0,0 +1,467 @@
+<!-- .slide: data-background="../content/images/title-slide.jpg" -->
+# Unit Testing
+
+---
+
+## How do I run a test in Angular? (10 minutes)
+
+---
+
+### The toolchain
+
+- Jasmine - the testing framework that ships with Angular
+- Karma - the test runner
+- Istanbul - generates coverage reports
+
+---
+
+### Running tests with the CLI and viewing output
+
+- `ng test` - lauches a browser for testing and watches for changes and compiling changes
+- `ng test --code-coverage` - puts a coverage report in the coverage/ directory
+
+---
+
+### Creating new elements and running tests in watch mode with the CLI
+
+Tests area automatically created by the CLI.
+
+- It puts test files alongside the file being tested so they are easier to find
+- It adds `.spec` to the test file names so Jasmine and the build system recognize them as tests
+
+1. Run `ng test --watch`
+2. Run `ng g c MyComponent`
+3. Open the test that is generated.
+
+---
+
+## How do I write tests? (10 minutes)
+
+---
+
+### Arrange, Act and Assert with Jasmine
+
+Note the following in the generated test:
+
+- `describe` explains what the test is
+- `beforeEach` runs code before each test
+- `it` explains the result we are expecting
+- `expect` formulates a scenario
+- `toBeTruthy` is a matcher that asserts the expected outcome
+
+---
+
+## How do I test a pipe?
```

Comment:
Just in case my last comment didn't come accross the right way, what I meant was that we need to explain why pipes are first in the content.  I was thinking we could put it in the slides, by saying something like "pipes are among the easiest things to test so...", but I don't like that either. Likely, the presenter would mention that before starting.